packages/phishing-controller/src/PhishingController.test.ts

```diff
@@ -2202,4 +2331,135 @@ describe('PhishingController', () => {
+  it('isBlockedRequest - should return false if the URL is in the whitelist', async () => {
+    const whitelistedHostname = 'example.com';
+
+    nock(PHISHING_CONFIG_BASE_URL)
+      .get(METAMASK_STALELIST_FILE)
+      .reply(200, {
+        data: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          eth_phishing_detect_config: {
+            allowlist: [whitelistedHostname],
+            blocklist: [],
+            fuzzylist: [],
+          },
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          phishfort_hotlist: {
+            blocklist: [],
+          },
+          tolerance: 0,
+          version: 0,
+          lastUpdated: 1,
+        },
+      })
+      .get(`${METAMASK_HOTLIST_DIFF_FILE}/${1}`)
+      .reply(200, { data: [] });
+
+    nock(CLIENT_SIDE_DETECION_BASE_URL)
+      .get(C2_DOMAIN_BLOCKLIST_ENDPOINT)
+      .reply(200, {
+        recentlyAdded: [],
+        recentlyRemoved: [],
+        lastFetchedAt: 1,
+      });
+
+    const controller = getPhishingController();
+    controller.bypass(formatHostnameToUrl(whitelistedHostname));
+    await controller.updateStalelist();
+    const result = controller.isBlockedRequest(
+      `https://${whitelistedHostname}/path`,
+    );
+
+    expect(result).toMatchObject({
+      result: false,
+      type: PhishingDetectorResultType.All,
+    });
+  });
+  it('isBlockedRequest - should return false if the URL is in the allowlist', async () => {
+    const whitelistedHostname = 'example.com';
+
+    nock(PHISHING_CONFIG_BASE_URL)
+      .get(METAMASK_STALELIST_FILE)
+      .reply(200, {
+        data: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          eth_phishing_detect_config: {
+            allowlist: [whitelistedHostname],
+            blocklist: [],
+            fuzzylist: [],
+          },
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          phishfort_hotlist: {
+            blocklist: [],
+          },
+          tolerance: 0,
+          version: 0,
+          lastUpdated: 1,
+        },
+      })
+      .get(`${METAMASK_HOTLIST_DIFF_FILE}/${1}`)
+      .reply(200, { data: [] });
+
+    nock(CLIENT_SIDE_DETECION_BASE_URL)
+      .get(C2_DOMAIN_BLOCKLIST_ENDPOINT)
+      .reply(200, {
+        recentlyAdded: [],
+        recentlyRemoved: [],
+        lastFetchedAt: 1,
+      });
+
+    const controller = getPhishingController();
+    await controller.updateStalelist();
+    const result = controller.isBlockedRequest(
+      `https://${whitelistedHostname}/path`,
+    );
+
+    expect(result).toMatchObject({
+      result: false,
+      type: PhishingDetectorResultType.Allowlist,
+    });
+  });
+  describe('PhishingController - bypass', () => {
+    let controller: PhishingController;
+
+    beforeEach(() => {
+      controller = getPhishingController();
+    });
+
+    it('should do nothing if the origin is already in the whitelist', () => {
```

Comment:
Shouldn't this test call `bypass` first, test the state, then call `bypass` again and check that the state hasn't increased?